shared/chat/inbox/container/remote.js

```diff
@@ -26,47 +17,57 @@ export type RemoteConvMeta = $Diff<
   {onSelectConversation: () => void}
 >
 
-// Sort by timestamp
-const getSortedConvMetas = Container.createSelector([getMetas], ([map, state]) =>
-  map
-    .sort((a, b) => b.timestamp - a.timestamp)
-    .slice(0, maxShownConversations)
-    .toList()
-    .map((m): RemoteConvMeta => {
-      const hasUnread = Constants.getHasUnread(state, m.conversationIDKey)
-      const styles = Constants.getRowStyles(m, false, hasUnread)
-      const participantNeedToRekey = m.rekeyers.size > 0
-      const _username = state.config.username || ''
-      const youNeedToRekey = !!participantNeedToRekey && m.rekeyers.has(_username)
-      return {
-        backgroundColor: Styles.globalColors.white,
-        channelname: m.channelname,
-        conversationIDKey: m.conversationIDKey,
-        hasBadge: Constants.getHasBadge(state, m.conversationIDKey),
-        hasResetUsers: !!m.resetParticipants && m.resetParticipants.size > 0,
-        hasUnread,
-        iconHoverColor: styles.iconHoverColor,
-        isFinalized: !!m.wasFinalizedBy,
-        isInWidget: true,
-        isMuted: m.isMuted,
-        isSelected: false,
-        // excluding onSelectConversation
-        participantNeedToRekey,
-        participants: m.teamname ? [] : Constants.getRowParticipants(m, _username).toArray(),
-        showBold: styles.showBold,
-        snippet: m.snippet,
-        snippetDecoration: m.snippetDecoration,
-        subColor: styles.subColor,
-        teamname: m.teamname,
-        timestamp: Constants.timestampToString(m),
-        usernameColor: styles.usernameColor,
-        youAreReset: m.membershipType === 'youAreReset',
-        youNeedToRekey,
-      }
-    }).toArray()
+// To cache the list
+const valuesCached = memoize((...vals) => vals.map(v => v))
+
+const metaMapToFirstValues = memoize(metaMap =>
+  metaMap
+    .partialSort(maxShownConversations, (a, b) => b.timestamp - a.timestamp)
+    .filter((_, id) => Constants.isValidConversationIDKey(id))
```

Comment:
I think this should go before `partialSort`

Reply:
i did this on purpose to not go through the entire list twice

Reply:
But the filter can result in removal of items, which will cause there to be fewer than `maxShownConversations`.

Reply:
in practice this never happens as the timestamps of the non valid conversations are zero so they won't ever be included if there are real convos. if there are real convos they'll show up first

Reply:
That's true right now, but it won't be true as we add more filters.

Reply:
what other filters would we add? the point of this exercise is to speed this thing up

Reply:
https://keybase.atlassian.net/browse/KBFS-3331

Reply:
I guess you could add the filter to the `partialSort` lambda function.

Reply:
if we need to filter beforehand we can but we don't in this code so it doesn't need to move. i prefer to not add complexity for some future case